BOJ/공항/김선필.cpp

```diff
@@ -0,0 +1,53 @@
+﻿#include <iostream>
+#include <vector>
+
+using namespace std;
+
+vector<int> par;
+
+int find(int u) {
+	return (u == par[u] ? u : par[u] = find(par[u]));
+}
+
+void merge(int u, int v) {
+	u = find(u);
+	v = find(v);
+
+	if (u == v) {
+		u = find(--u);
+	}
+
+	par[v] = u;
+}
+
+int main() {
+	ios_base::sync_with_stdio(false);
+	cin.tie(nullptr);
+
+	int G, P;
+	cin >> G >> P;
+	
+	par.resize(G + 1);
+	for (int i = 1; i <= G; i++) {
+		par[i] = i;
+	}
+
+	vector<int> g(P);
+	for (int i = 0; i < P; i++) {
+		cin >> g[i];
+	}
+
+	int count = 0;
+	for (int i = 0; i < P; i++) {
+		if (find(g[i]) == 0) {
+			break;
+		}
+
+		count++;
+		merge(g[i] - 1, g[i]);
+	}

```

Comment:
유의미한 시간 감소일지는 모르겠는데, 입력을 받음과 동시에 연산을 한다면 시간을 줄일 수 있을것 같습니다.
```suggestion
	int count = 0;
	for (int i = 0; i < P; i++) {
	       cin >> g[i];

		if (find(g[i]) == 0) {
			break;
		}

		count++;
		merge(g[i] - 1, g[i]);
	}

```

Reply:
아~ 이렇게 하면 중간에 입력받지 못하는 게 생겨서 문제가 될 줄 알았는데, 수정해서 제출해 보니 문제가 없네요! 좋은 의견 감사합니다😄